xbmc/platform/posix/MemUtils.cpp

```diff
@@ -69,10 +69,10 @@ void GetMemoryStatus(MemoryStatus* buffer)
   uint64_t physmem;
   size_t len = sizeof physmem;
   std::array<int, 2> mib =
-  {
+  {{
     CTL_HW,
     HW_MEMSIZE,
-  };
+  }};
```

Comment:
There should at least added a doxygen comment about the compiler bug, so the workaround can be found and removed later.
You could also disable the warning for this lines via a pragma.

```
#if defined(__apple_build_version__) && __apple_build_version__ < 10000000
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wmissing-braces"
#endif
  std::array<int, 2> abc =
  {
    CTL_HW,
    HW_MEMSIZE,
  };
#if defined(__apple_build_version__) && __apple_build_version__ < 10000000
#pragma clang diagnostic pop
#endif
```